fineftp-server/src/filesystem.cpp

```diff
@@ -209,20 +209,20 @@ namespace Filesystem
       "Dec"
     };
 
-    // Use strftime for the day and year / time
-    char date[80];
+    std::stringstream date;
+
     if (file_year == current_year)
     {
       // We are allowed to return the time!
-      strftime(date, sizeof(date), " %e %R", &file_timeinfo);
+      date << " " << file_timeinfo.tm_mday << " " << file_timeinfo.tm_hour << ":" << file_timeinfo.tm_min;
     }
     else
     {
-      // We must not return the time, only the date :(
-      strftime(date, sizeof(date), " %e  %Y", &file_timeinfo);
+      static constexpr auto tm_year_base_year = 1900;
+      date << " " << file_timeinfo.tm_mday << " " << ( file_timeinfo.tm_year + tm_year_base_year );
```

Comment:
Please change, so this outputs the date with leading space in case of a single-digit day, e.g. with [std::setw](https://en.cppreference.com/w/cpp/io/manip/setw) and [std::setfill](https://en.cppreference.com/w/cpp/io/manip/setfill)

```
<day> ::= <digit> <digit>
        |   SPC   <digit>

<year> ::= <digit> <digit> <digit> <digit>
```

https://files.stairways.com/other/ftp-list-specs-info.txt